### PR TITLE
Fix issue 579: Bug in time-averaging and instantaneous ABL statistics

### PR DESCRIPTION
**Pull request by lawrenceccheung**
**Pull-request type:**
- [X] Bug fix 
- [ ] Documentation update
- [ ] Feature enhancement

## Description of the pull-request

Fix problems encountered in issue [579](https://github.com/Exawind/nalu-wind/issues/579):
- Time averaged temperature fluxes in netCDF statistics files were incorrectly computed at t=0.
- Time averaged temperature variances in netCDF statistics files were incorrectly computed at t=0.
- Instantaneous SFS statistics were incorrectly writen to netCDF file.

Adds a new regression test (`ablNeutralStat`) which tests for some of these statistics quantities.  Note that `ablNeutralStat` only checks for the *internal consistency* of the time-average and instantaneous statistics, but doesn't compare the netCDF statistics to a previously computed "gold" solution.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [X] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [X] Linux
    - [ ] MacOS
  - Compilers 
    - [ ] GCC
    - [ ] LLVM/Clang
    - [X] Intel compilers
    - [ ] NVIDIA CUDA
- [X] Compiles without warnings
- [X] Passes all unit tests
- [X] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [ ] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [X] New regression tests exercising the new code

### CODE REVIEW

File: reg_tests/test_files/ablNeutralStat/README [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+case.  The case should be run for at least 16,000 s to acheive a level
+of quasi equlibrium.  We recommend running this case with 10 m uniform
+resolution in all directions (it is set up for uniform 40 m resolution
+for this regression test).  The stability is neutral with no heat flux
+at the ground, but the boundary layer is capped with a stable inversion
+layer.  Because of constant density used here, because we do not include
+pressure work term in the enthalpy equation, and becuase we use Boussinesq
+buoyancy, temperature is potential temperature.
+
+- To build the mesh, use NaluWindUtils abl_mesh with nalu_abl_mesh.yaml
```
Comment (sayerhs) [inline reg_tests/test_files/ablNeutralStat/README:22]: `NaluWindUtils` was renamed to `wind-utils` when we imported it into `exawind` org.

File: reg_tests/test_files/ablNeutralStat/nalu_preprocess.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# -*- mode: yaml -*-
```
Comment (sayerhs) [inline reg_tests/test_files/ablNeutralStat/nalu_preprocess.yaml:1]: Why do we need this file?

File: reg_tests/test_files/ablNeutralStat/ablNeutralStat.i.in [hunk 3]
```diff
@@ -0,0 +1,298 @@
+Simulations:
```
Comment (sayerhs) [inline reg_tests/test_files/ablNeutralStat/ablNeutralStat.i.in:1]: Why do we need this file?

File: reg_tests/test_files/ablNeutralStat/nalu_abl_mesh.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+nalu_abl_mesh:
```
Comment (sayerhs) [inline reg_tests/test_files/ablNeutralStat/nalu_abl_mesh.yaml:1]: Not needed?

File: reg_tests/test_files/ablNeutralStat/passfail.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        self.u   = velocity[:,:,0]
+        self.v   = velocity[:,:,1]
+        self.w   = velocity[:,:,2]
+
+        # Temperature
+        temperature = self.abl_stats.variables['temperature']
+
+    def get_latesttime(self, field='velocity', index=0, scalar=False, zeroD=False):
+        '''
+        Provide field time average
```
Comment (sayerhs) [inline reg_tests/test_files/ablNeutralStat/passfail.py:44]: Cut/paste issue?

File: reg_tests/test_files/ablNeutralStat/catalyst.txt [hunk 6]
```diff
@@ -0,0 +1,24 @@
+begin catalyst
```
Comment (sayerhs) [inline reg_tests/test_files/ablNeutralStat/catalyst.txt:1]: Remove

File: reg_tests/test_files/ablNeutralStat/README (hunk 1 above)
Comment (lawrenceccheung) [reply reg_tests/test_files/ablNeutralStat/README]: Changed the name in the new README.

File: reg_tests/test_files/ablNeutralStat/nalu_abl_mesh.yaml (hunk 4 above)
Comment (lawrenceccheung) [reply reg_tests/test_files/ablNeutralStat/nalu_abl_mesh.yaml]: deleted

File: reg_tests/test_files/ablNeutralStat/catalyst.txt (hunk 6 above)
Comment (lawrenceccheung) [reply reg_tests/test_files/ablNeutralStat/catalyst.txt]: deleted

File: reg_tests/test_files/ablNeutralStat/passfail.py (hunk 5 above)
Comment (lawrenceccheung) [reply reg_tests/test_files/ablNeutralStat/passfail.py]: Comment has been changed.

**Comment by lawrenceccheung** [comment]
I cleaned up the unused files in the latest commit.

Lawrence

File: reg_tests/test_files/ablNeutralStat/nalu_preprocess.yaml (hunk 2 above)
Comment (lawrenceccheung) [reply reg_tests/test_files/ablNeutralStat/nalu_preprocess.yaml]: Deleted

**Pull request merged**